### PR TITLE
Minor tweaks including json support for render/renderFile

### PR DESCRIPTION
**Pull request by JohnAD**
The PR does three things:

1. For the latest versions of nim (incl 1.0.4) it removes the compiler errors/warnings about `kind` changes.
2. It makes two unit tests invoked by `nimble test` more consistent with nimble recommendations.
3. It adds support for passing JSON directly to `render` and `renderFile`. So, instead of:

```nim
   var jc = newJObject()
   # ... stuff happens to jc
   let c = newContext(jc)
   let myResult = render(tmplate, c)
```

you can simply...

```nim
   var jc = newJObject()
   # ... stuff happens to jc
   let myResult = render(tmplate, jc)
```

Since `json` is part of the standard library and is heavily used by web site building (a common use-case of templating), I think this makes sense. My next Nim-website-building video will go through creating HTML views and will feature `moustachu`. It would be great if I could avoid the conversion step to keep things simple.

ref to video playlist:
https://www.youtube.com/playlist?list=PL6RpFCvmb5SGw7aJK1E4goBxpMK3NvkON

### CODE REVIEW

File: src/moustachu.nim [hunk 1]
```diff
@@ -65,9 +66,9 @@ proc parallelReplace(str: string,
     result = result.replace(sub[0], sub[1])
 
 proc render(tmplate: string, contextStack: seq[Context], partialsDir="."): string =
-  ## Take a mustache template `tmplate`, an evaluation Context `c`, and an optional
-  ## path `partialsDir` (by default set to the current directory) which may contain partial files
-  ## and return the rendered string. This is the main procedure.
+  # Take a mustache template `tmplate`, an evaluation Context `c`, and an optional
+  # path `partialsDir` (by default set to the current directory) which may contain partial files
+  # and return the rendered string. This is the main procedure.
```
Comment (fenekku) [inline src/moustachu.nim]: We can keep the double ## (documentation comments), right?

File: tests/config.nims [hunk 2]
```diff
@@ -0,0 +1 @@
+switch("path", "$projectDir/../src")
```
Comment (fenekku) [inline tests/config.nims:1]: neat!

File: tests/test_json.nim [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+suite "test json input":
+  test "from string template":
+    var j = newJObject()
+    j["string"] = newJString("hello")
+    j["int"] = newJInt(3)
+    j["float"] = newJFloat(3.14)
+    let expected_result = "s=hello,i=3,f=3.14"
+    check render(objTemplate, j) == expected_result
+  test "from file":
```
Comment (fenekku) [inline tests/test_json.nim:17]: minor: new line between the `test` blocks

**Review by fenekku — APPROVED** [review]
Thanks for this. Two minor things and also bump the version to 0.14.0. Then I will merge!

File: src/moustachu.nim (hunk 1 above)
Comment (JohnAD) [reply src/moustachu.nim]: I'm under the impression that docgen comments are for public procedures. If I restore it to documentation comments, I will mention that it is private. Or, I could make it public. I don't see any harm in making it public. Thoughts?

File: tests/test_json.nim (hunk 3 above)
Comment (JohnAD) [reply tests/test_json.nim:17]: Will add

**Comment by JohnAD** [comment]
I made the two changes and bumped the version to 0.14.0. To keep the doc comment consistent, I made the related `render` function public. Let me know if you want me to undo that.

I also found two more compiler warning messages and cleaned those up while I was at it. Not sure how I missed them earlier. Everything compiles without warnings or errors now. This includes the latest stable nim: 1.0.6.

Thanks for looking at this!

**Pull request merged**